v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -166,29 +135,91 @@ export async function main(cliArguments: string[]) {
 
     console.log("Error running the task:", error.message);
 
-    if (showStackTraces) {
+    if (hhGlobalParams.showStackTraces) {
       console.log("");
       console.error(error);
     }
   }
 }
 
-function parseGlobalArguments(
-  _globalParamsIndex: GlobalParameterMap,
-  _cliArguments: string[],
-  _usedCliArguments: boolean[],
-): Partial<GlobalArguments> {
-  // TODO: Parse the global arguments
-  // - Parse the global params, skipping the processed entries
-  // - At each stage validate the value according to its type
-  // - If a global param is boolean and its default is false, its value is not
-  //   necessary. That means that if the immediatly next value is `"true"` or
-  //   `"false"`, we should use it, but if it's not we just don't, and consider
-  //   its value `true`.
-  // - Mark all the used entries in usedCliArguments
-  // - Do not resolve defaults here, the HRE does that.
-  // - Return the values
-  return {};
+export async function parseInitialHardhatParameters(
+  cliArguments: string[],
+  usedCliArguments: boolean[],
+) {
+  let configPath: string | undefined;
+  let showStackTraces: boolean = false; // true if ci
+  let help: boolean = false;
+  let version: boolean = false;
+
+  for (let i = 0; i < cliArguments.length; i++) {
+    const arg = cliArguments[i];
+
+    if (arg === "--config") {
+      usedCliArguments[i] = true;
+
+      if (configPath !== undefined) {
+        throw new HardhatError(HardhatError.ERRORS.ARGUMENTS.DUPLICATED_NAME, {
+          name: "--config",
+        });
+      }
+
+      if (
+        usedCliArguments[i + 1] === undefined ||
+        usedCliArguments[i + 1] === true
+      ) {
+        throw new HardhatError(
+          HardhatError.ERRORS.ARGUMENTS.MISSING_CONFIG_FILE,
+        );
+      }
+
+      configPath = cliArguments[i + 1];
+      i++;
+
+      usedCliArguments[i] = true;
+      continue;
+    }
+
+    if (arg === "--show-stack-traces") {
+      usedCliArguments[i] = true;
+      showStackTraces = true;
+      continue;
+    }
+
+    if (arg === "--help") {
+      usedCliArguments[i] = true;
+      help = true;
+      continue;
+    }
+
+    if (arg === "--version") {
+      usedCliArguments[i] = true;
+      version = true;
+      continue;
+    }
+  }
+
+  return { configPath, showStackTraces, help, version };
+}
+
+export async function parseGlobalArguments(
+  globalParamsIndex: GlobalParameterMap,
+  cliArguments: string[],
+  usedCliArguments: boolean[],
+): Promise<Partial<GlobalArguments>> {
+  const globalArguments: Partial<GlobalArguments> = {};
+
+  const parameters = new Map(
+    Array.from(globalParamsIndex, ([key, value]) => [key, value.param]),
+  );
```

Comment:
```suggestion
  const parameters = new Map(
    [...globalParamsIndex].map(([key, value]) => [key, value.param]),
  );
```

Reply:
[Done in this commit](https://github.com/NomicFoundation/hardhat/pull/5334/commits/52d98a27918466f91da88df82eccaba54e02eac3)